src/components/tile/TileWithButton.tsx

```diff
@@ -55,25 +56,27 @@ export function TileWithButton({
       >
         {children}
       </View>
-      <View
-        style={[
-          styles.button,
-          mode === 'spacious' ? styles.spaciousButton : styles.compactButton,
-        ]}
-        testID={testID + 'Button'}
-      >
-        <ThemeText
-          style={styles.buttonText}
-          type={mode === 'spacious' ? 'body__primary' : 'body__tertiary'}
+      {onPress !== undefined && (
+        <View
+          style={[
+            styles.button,
+            mode === 'spacious' ? styles.spaciousButton : styles.compactButton,
+          ]}
+          testID={testID + 'Button'}
         >
-          {buttonText}
-        </ThemeText>
-        <ThemeIcon
-          size={mode === 'spacious' ? 'normal' : 'xSmall'}
-          svg={buttonSvg}
-          fill={color.outline.text}
-        />
-      </View>
+          <ThemeText
+            style={styles.buttonText}
+            type={mode === 'spacious' ? 'body__primary' : 'body__tertiary'}
+          >
+            {buttonText}
+          </ThemeText>
+          <ThemeIcon
+            size={mode === 'spacious' ? 'normal' : 'xSmall'}
+            svg={buttonSvg}
+            fill={color.outline.text}
+          />
+        </View>
+      )}
```

Comment:
Can the button be disabled instead of removing it? I don't think it makes sense to have a `TileWithButton` without the button. If they want an info box without button they can show a `BorderedInfoBox` instead.

Also by skimming the code, maybe we doesn't need support for undefined `onPress` at this time.